### PR TITLE
Unify map parsers

### PR DESCRIPTION
**Pull request by jkbmrz**
I'm proposing to unify parsing of models that output `2D maps`, such as `depth maps` (e.g. DepthAnything), `density maps` (e.g. DM-Count), `heat maps`, and similar. It removes redundant code and simplifies message creation (e.g. omitting the model-specific parameters like `depth_type`, `depth_limit `etc.).

Some open questions:

1. Should we extend the unification also to `segmentation maps` or do we want to keep the two separate (differently from others, segmentation maps consist of int values instead of floats)?
2. There is no native `DepthAI` message for 2D float arrays. Previously we had it coded it as a `dai.ImgFrame.Type` and were scaling it to specific ranges so we could later get out the original float values. But this was messy so a new custom Map2D message is introduced here. In our recent discussion, however, it was advised to refrain from using a custom type. Thus, let's discuss here if a custom message is appropriate here or should we try basing it on some native message?

### CODE REVIEW

**Review by kkeroo — APPROVED** [review]
LGTM otherwise.

**Comment by kkeroo** [comment]
> Some open questions:
> 
> 1. Should we extend the unification also to `segmentation maps` or do we want to keep the two separate (differently from others, segmentation maps consist of int values instead of floats)?
> 2. There is no native `DepthAI` message for 2D float arrays. Previously we had it coded it as a `dai.ImgFrame.Type` and were scaled it to specific ranges so we could later on get out the original float values. But this was messy so a new custom Map2D message is introduced here. In our recent discussion, however, it was advised to refrain from using a custom type. Thus, let's discuss here if a custom message is appropriate here or should we try basing it on some native message?

1. I think we can leave segmentation in dai.ImgFrame for now because segmentation masks can be easily represented with ImgFrame

2. For me using 2D map msg makes sense. It also simplifies the parsers and allows merging them together (MonocularDepthParser and DM-Count parser). However, one idea came to my mind. Since we want to use DAI native msgs (ImgFrame and NNData) so we can send them to device and back we can maybe combine both approaches. Instead of creators functions already creating custom msg type we can have encoder/decoder functions. Encoder function will encode the parser's results into NNData or some other DAI native msg and decoder will decode this msg into user-friendly msg. 

Parser node
```
msg: dai.NNData = encode_depth_msg()
self.out.send(msg)
```

Main pipeline
```
while pipeline.isRunning():
    ...
    msg = queue.get()
    depth_map: DepthMap = decode_depth_msg(msg)
```

So instead of sending custom msg out of parser we send out dai native msg and decode the msg into custom msg class (e.g. Keypoints) in the pipeline.

**Comment by klemen1999** [comment]
1/ Agree, we can keep segmentation separate.
2/ Let's go with the the custom message type for now (`Map2D`) and when we align with DAI team regarding support (or lack of it) for custom message types we can address this in a separate PR (could potentially need a bigger refactor). If custom types will not be supported then idea of encoder/decoder pattern make sense.

File: depthai_nodes/ml/messages/creators/map.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import depthai as dai
+import numpy as np
+
+from ...messages import Map2D
+
+
+def create_map_message(map: np.ndarray, min_max_scaling: bool = False) -> dai.ImgFrame:
```
[the remark this answers is not in the document]
Comment (klemen1999) [reply depthai_nodes/ml/messages/creators/map.py:7]: @jkbmrz ^ (in case you've missed the ping :smiley: )

**Review by klemen1999 — APPROVED** [review]
LGTM

File: depthai_nodes/ml/messages/creators/map.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jkbmrz) [reply depthai_nodes/ml/messages/creators/map.py]: Thanks, added in [9aa4881](https://github.com/luxonis/depthai-nodes/pull/57/commits/9aa48819c68c9c4e86372774998545755403a71d).

**Pull request merged**